tutorials/decryption.md

```diff
@@ -0,0 +1,93 @@
+
+
+## Configuring Image Decryption for **Node** key model
+
+In order to set up image decryption support, modify `/etc/crio/crio.conf` as follows:
+
+```toml
+# decryption_keys_path is the path where the keys required for
+# image decryption are located
+decryption_keys_path = "/etc/crio/keys/"
+
+```
+`decryption_keys_path` should be the path where `CRI-O` can find the keys required for image decryption.
+
+After modifying this config, you need to restart the `CRI-O` service.
+
+Alternatively, if you are starting the `CRI-O` from the command line, the argument `--decryption-keys-path` can be provided pointing to the folder that contains required decryption keys.
+
+## Verification of the Decryption Capabilities
+
+Although the latest master branch of the `docker/distribution` registry supports encrypted images, many popular public registries such as docker hub or `quay.io` don't support encrypted images yet.
+
+For the easy verification of the image decryption capabilities, we are hosting a test image at,
+
+`us.icr.io/ecins/encrypted_image:encrypted`
+
+Go ahead and try to download this image using the read-only credentials given below,
+
+```sh
+crictl -r unix:///var/run/crio/crio.sock pull --creds iamapikey:x_egxeGnaXi4GfKMbp0pYw0iErUAIjn5uYQIHTZ2RKof us.icr.io/ecins/encrypted_image:encrypted
+```
+
+Since we haven't provided `CRI-O` the access to the private key required to decrypt this image you should see a failure like this on your console,
+
+```sh
+FATA[0010] pulling image failed: rpc error: code = Unknown desc = Error decrypting layer sha256:ecbef970c60906b9d4249b47273113ef008b91ce8046f6ae9d82761b9ffcc3c0: missing private key needed for decryption
+```
+
+This is intended behavior and proof that indeed the encrypted images cannot be used without having access to the correct private key.
+
+The image can be decrypted using following key,
+```sh
+-----BEGIN RSA PRIVATE KEY-----
+MIICXAIBAAKBgQDoJBuK1hQ5aCbF93uE6jzRm8v5icUNFL5j+DO9hnM5j/8XFTzp
+40N2M2/ObLf2qwmWSivwj5LJR/+5ceS8jqVBAcJpckwOXupu3A5o4KgJo15s6v57
+4+0wfraNJ/OapqBc7lGFBsj+XwdmegwYYqy41DnYNSzYS4Mov+v7RI014wIDAQAB
+AoGALCuiqfouAvZUWlrKv/Gp/OA+IY8bVW/bAj6Z6bgJeKxzhzrdSkuZ7IXBAnAh
+WOgWfOhEEBPhhDcU635GXbJusuD/bLBJPOTxiwCFazffm8zVGSQCndfTVxgCM4hn
++5bH6o/cSGQ4E6SLJQeEr8y/J0bMlNMkOco9F1FL1ZgwXGECQQD9/mDwWLJjbdEa
+jmGtoPspGz80XDb1jRI09jKDXB826/cBUD+X/P50aTkU+XSJXVfa5F6zhzf/O7C8
+07bVnn2pAkEA6fmJ+Jx/Cupy7jRHzIdKAN/7T9QJBIXVDZLz5ulFWLjYkNotpkxk
+f0ZSIOvlD7vv5lOifRFivd680XjxIATWqwJARJ35QFUl9DiRuhPnDYok8Cj9PT8A
+VfwDhC1S3iv//s1mkINGeuANOhPHKQRvWEDQYEE72FJabWiJyamEhldn6QJAFjLw
+3j+q5hQ8d1FKhqNHaDHYHEjX2jAAeNs6fOwhAjv3gDbTIfYZiuHXJPx8rTN9nXLN
+9ePSZIVfkNhSuGD9JQJBAI+mobcxj7WkdLHuATdAso+N89Yt7xHoG49c8gz81ufP
+vvLPtYytL4ftpiVO3fTfPP90ze8qYPiNaFqMHYDkQ+M=
+-----END RSA PRIVATE KEY-----
+```
+
+Please save this key in the folder `/etc/crio/keys` with the name of your choice.
```

Comment:
I'd feel better if we added some type of note saying that in general, sharing private keys is a very, very bad idea.

Reply:
@TomSweeneyRedHat Thanks. I added following text at the bottom of that tutorial.
```
Please note that the confidentiality provided by the encrypted images could get compromised if the private keys are accessed by authorized and/or unintended entities. Also, in case of loss of private key, there is no way to access the contents of the encrypted image rendering it completely unusable. Hence, it's extremely important to keep the private keys securely and safely.
```